### DISCUSSION TITLE
Hash routes?

### DISCUSSION THREAD

**Discussion by fnick851** [pinecone-router/router#18] [Q&A] [answered]
Since Alpine.js is typically used together with a server side framework, is there an option to enable hash routes so that the route doesn't conflict with the backend router? 

For example, the Vue router has a hash mode: https://router.vuejs.org/guide/essentials/history-mode.html#hash-mode
React has it too: https://reactrouter.com/en/main/router-components/hash-router

Thanks.

**Comment by rehhouari** [comment, marked as answer]
hey have you tried using [this settings option](https://github.com/pinecone-router/router#settings)?
does it do what you need?

**Reply by fnick851** [reply]
Thank you! I didn't see that somehow.

**Comment by fnick851** [comment]
@rehhouari 
When I set `window.PineconeRouter.settings.hash = true`, if I go to say https://example.com/folder/page.asp, the URL gets automatically changed to https://example.com/folder/page.asp#example.com/folder/page.asp

This causes the hash mode unusable at all. Could you give a working example of a hash mode router? Thanks.

**Comment by rehhouari** [comment]
hey sorry for the inconvenience! it's true what you mentioned i dont know why it automatically added the path to the hash. I removed that part and it should work now? make sure you're on version 3.1.0

let me know if you have any more questions.  ^^

**Reply by berndca** [reply]
Hi, I'm trying to migrate from alpinejs-router to pinecone-router. I don't seem to be able to get the hash mode to work. I have included  the settings option according to the documentation.  The only thing I can see is that a trailing '#' is tolerated, but that's not hash mode. For instance for my login route I expect an URL like http://127.0.0.1:5500/#/log-in. When I enable hash mode in your demo and try the "hello world" route I get http://127.0.0.1:5500/hello/world instead of http://127.0.0.1:5500/#/hello/world. Please advise. Thanks

**Reply by rehhouari** [reply]
hey @berndca sorry about the demo it used an older version instead of 3.x ! it's updated now and it have an [option here](https://github.com/rehhouari/pinecone-example/blob/80daad0b4da3e9d030e4c7b9c49d51345f716a90/index.html#L74) to test it or open the [demo site](https://pinecone-example.vercel.app) then set `window.PineconeRouter.settings.hash = true` in console then click on one of the buttons without reloading it should work.
hopefully in your demo you had the pinecone-router version set to 2.x as well?

**Reply by berndca** [reply]
I installed a local copy of the demo on my machine and upgraded pinecone-router to the latest version. I checked in the debugger, the version is 3.1.0. To be honest reloading and bookmarking is a very large part of why I need hash support. The hash needs to be BETWEEN the base URL, e.g. http://127.0.0.1:5500 and the route e.g. /hello/world = http://127.0.0.1:5500/#/hello/world. The hash should be inserted by the router, i.e. all URL's generated by the router should have the hash in the correct position. I was unable to get this behavior with pinecone-router 3.1.0. Am I missing anything? Thanks

**Reply by rehhouari** [reply]
im sorry i cant figure out where its wrong i just followed those exact steps and reloading works and the hash is between the route aswell thats weird

**Reply by berndca** [reply]
Thanks! I started over with a new copy and it's working now.

**Reply by rehhouari** [reply]
glad its working! and sorry for the inconvenience ^^

**Reply by fnick851** [reply]
@rehhouari No need to feel sorry. This is open source and you are not responsible for solving our problems. Of course we appreciate your efforts making such a useful little library. Glad you could help but just don't get burned out by the "demands" from the community. You are not obliged to do these to begin with. 

I just have seen too many open source stories where the maintainer eventually abandoned a project due to stress of not being able to meet what the community demands. You are doing this for free and it shouldn't be a burden.
